nix/download.tt

```diff
@@ -4,23 +4,30 @@
 
 <p>The latest and recommended version of Nix is
 <strong>[%latestNixVersion%]</strong>. The quickest way to install it
-on Linux and Mac OS X is to run the following in a shell (as a user
+on Linux and macOS is to run the following in a shell (as a user
 other than <tt>root</tt>):
 
-<pre><span class="muted">$</span> curl <a href="[%root%]nix/install">https://nixos.org/nix/install</a> | sh</pre>
+<pre>
+<span class="command">curl -o install-nix-[%latestNixVersion%] https://nixos.org/nix/install</span>
+<span class="command">echo "[%nixInstallHashsha256%]  install-nix-[%latestNixVersion%]" | (shasum -p 256 || sha256sum -c) 2> /dev/null</span>
+<span class="command">sh install-nix-[%latestNixVersion%]</span>
+</pre>
```

Comment:
Will there be a case when `sha256sum` is available but `shasum` is not (or vice-verse)? Or, would there be a package difference that would cause `shasum -p 256` to generate a _different_ hash from `sha256sum -c`?

I ask because if not for both, which I think would be the case, we could actually lessen this to only include `sha256sum -c` for a shorter line overall (or just `shasum -p 256` if we're worried that `sha256sum` isn't avialable). That, or if precedence is for `sha256sum`, perhaps flip the order to `(sha256sum -c || shasum -p 256)`.